pkg/sampling/encoding_test.go

```diff
@@ -0,0 +1,304 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sampling
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+// must panics when the error is non-nil and returns the second
+// generic argument.  this does not call require.NoError() in order to
+// use a one-line test calling convention, meaning `must(functionCall())`
+// ensures there is no error and returns the first argument.
+//
+// To do the same with NoError() means rewriting the expression with
+// a two-line statement:
+//
+//	value, err := functionCall()
+//	require.NoError(t, value)
+//
+// and that is why this function exists.
+func must[T any](t T, err error) T {
+	if err != nil {
+		panic(err)
+	}
+	return t
+}
+
+// mustNot panics when the error is nil.  this does not call
+// require.Error() in order to use a one-line test calling
+// convention, meaning `mustNot(functionCall())` ensures there is an
+// error without requiring a separate variable assignment.
+//
+// To do the same with Error() means rewriting the expression with
+// a two-line statement:
+//
+//	_, err := functionCall()
+//	require.Error(t, value)
+//
+// and that is why this function exists.
```

Comment:
```suggestion
```

I think you've added more lines of code than you saved with these functions if you spend this much time explaining them. But I don't have a problem with making tests easier to write, so go for it.